src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskExtensions.cs

```diff
@@ -26,12 +27,16 @@ internal static class TaskExtensions
         /// </exception>
         public static async Task<bool> TryWaitForCompleteOrTimeout(this Task task, int millisecondsTimeout)
```

Comment:
💡 Consider replacing this method with `WithTimeout` from vs-threading:

https://github.com/Microsoft/vs-threading/blob/34c6eae89e0514ca6c6e72d411629cdb4cd88d2e/src/Microsoft.VisualStudio.Threading/TplExtensions.cs#L73

It's not a direct replacement since `WithTimeout` throws a `TimeoutException`, but it may work for what you need in practice.

Reply:
I wouldn't use it here, because it's not a direct replacement.

Reply:
Thanks for the tip @sharwell but I think the exception makes it not a great fit here. Will keep it in mind for the future.